sim/common/cata/other_effects.go

```diff
@@ -762,6 +755,24 @@ func init() {
 				},
 			})
 
+			character.AddDynamicDamageTakenModifier(func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+				shield := shieldSpell.SelfShield()
+				if shield.IsActive() && result.Damage > 0 {
+					absorbedDamage := min(currentShield, result.Damage)
+					result.Damage -= absorbedDamage
+					currentShield -= absorbedDamage
+					shield.Aura.SetStacks(sim, int32(currentShield))
+
+					if sim.Log != nil {
+						character.Log(sim, "Loom of Fate absorbed %.1f damage", absorbedDamage)
+					}
+
+					if currentShield <= 0 {
+						shield.Deactivate(sim)
+					}
+				}
+			})
```

Comment:
Since we're now using this logic multiple times for trinkets, as well as for Savage Defense, you could consider adding a helper utility in core for it. For example,

```
func (character *Character) NewDamageAbsorptionAura(auraLabel string, actionID ActionID, duration time.Duration, shieldStrengthPtr *float64) *Aura {}
```

Reply:
Thinking about this more, we can even avoid the awkwardness of passing a pointer to the shield strength value if we embed it in a struct. For example:

```
type ShieldStrengthCalculator func(unit *Unit) float64

type DamageAbsorptionAura struct {
    *Aura
    ShieldStrength float64
    FreshShieldStrengthCalculator ShieldStrengthCalculator
}

func (aura *DamageAbsorptionAura) Activate(sim *Simulation) {
    aura.Aura.Activate(sim)
    aura.ShieldStrength = aura.FreshShieldStrengthCalculator(aura.Unit)
}
```

Then we could have the helper be constructed as

```
func (character *Character) NewDamageAbsorptionAura(auraLabel string, actionID ActionID, duration time.Duration, calculator ShieldStrengthCalculator) *DamageAbsorptionAura
```

The downstream code that calls this would then hold onto that returned pointer, and would simply have to call `Activate()` on it as part of either the trinket activation spell or the `ProcTrigger` handler, depending on how the shield is triggered.